utils/tagged/src/lib.rs

```diff
@@ -40,7 +42,9 @@ impl<T> Tagged<T> {
     ///
     /// - `T` must have an alignment of at least 2.
     /// - `tag` must fit inside `usize::BITS - 1` bits
-    pub(crate) const fn from_tag(tag: usize) -> Self {
+    #[inline]
```

Comment:
Had an idea. What if we moved the assertion to compile time? Should remove the alignment requirement from the docs, and that way you cannot create invalid tagged pointers:
```rust
    const ALIGNMENT_CHECK: () = assert!(align_of::<T>() >= 2);

    /// Creates a new, tagged `Tagged` pointer from an integer.
    ///
    /// # Requirements
    ///
    /// - `tag` must fit inside `usize::BITS - 1` bits
    #[inline]
    #[must_use]
    pub const fn from_tag(tag: usize) -> Self {
        let _ = Self::ALIGNMENT_CHECK;
        let addr = (tag << 1) | 1;
        // SAFETY: `addr` is never zero, since we always set its LSB to 1
        unsafe { Self(NonNull::new_unchecked(ptr::without_provenance_mut(addr))) }
    }
```

Reply:
Nice suggestion! :)